zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -208,13 +238,44 @@ private boolean parseSource(HttpMessage message, Source source, int depth, Strin
             resourcesfound |= processAttributeElement(message, depth, baseURL, el, "src");
         }
 
+        // Process Table elements
+        elements = source.getAllElements(HTMLElementName.TABLE);
+        for (Element el : elements) {
+            resourcesfound |= processAttributeElement(message, depth, baseURL, el, "background");
+        }
+
+        // Process TD elements
+        elements = source.getAllElements(HTMLElementName.TD);
+        for (Element src : elements) {
+            resourcesfound |= processAttributeElement(message, depth, baseURL, src, "background");
+        }
+
+        // Process Video elements
+        elements = source.getAllElements(HTMLElementName.VIDEO);
+        for (Element el : elements) {
+            boolean srcFound = processAttributeElement(message, depth, baseURL, el, "src");
+            resourcesfound |= srcFound;
+            resourcesfound |= processAttributeElement(message, depth, baseURL, el, "poster");
+            if (!srcFound) {
+                Source videoSource = new Source(el.getContent().toString());
```

Comment:
Why parse it again?

Reply:
It is not the same thing: here we are getting to body of the video tag to parse all the source elements